### DISCUSSION TITLE
Game shouldn't progress while holding setting sliders

### DISCUSSION THREAD

**Discussion by monochrome22** [ppy/osu#23347] [Ideas]
### Type

Game behaviour

### Bug description

Similar thing: https://github.com/ppy/osu/issues/6121

Found this when I was trying to adjust background dim during replay and the replay ended. but the adjusting remained.

The map won't start if you're adjusting sliders at the gameplay intro screen, but it will start if you're just holding the slider. Replay will end even if you're adjusting sliders.

It's worth considering to not start gameplay even when a slider, or anything with a tooltip for that matter, is just being hovered. For example, sometimes I hover the Difficulty Adjust mod icon in gameplay intro screen to double check what settings I set, but the gameplay starts while I'm reading.

### Screenshots or videos

https://user-images.githubusercontent.com/122969285/235352875-d7605275-5981-4208-bec6-9131f5e6a85f.mp4

### Version

2023.419.0-lazer

### Logs

Not necessary

**Comment by bdach** [comment]
I'm downgrading this to a discussion because this is a subjective UX issue.

It already is the case that the game checks whether you are hovering over a UI element when determining whether to progress to gameplay. However, that logic also has an idle timeout of 750ms, after which time it is determined that the user isn't doing anything worthwhile anymore and the game is allowed to progress to gameplay:

https://github.com/ppy/osu/blob/6a04708a7e9801949c6c7ac7ddf6a4d7fa0835e5/osu.Game/Screens/Play/PlayerLoader.cs#L102-L108

Whether the timeout should be increased, or the idle detection should be improved, is up for discussion, but currently this looks more or less to be working as intended.

**Reply by monochrome22** [reply]
My suggestion of not starting gameplay while hovering was only a sidenote of the main issue. The fact that replay ends or gameplay starts while holding or adjusting the slider is obviously broken. This should be converted back to an issue and I'll edit out the hovering part if it's out of scope.

This is clearly broken UX:

https://user-images.githubusercontent.com/122969285/235353760-b05e6ea5-7a89-412a-a223-de1bec179306.mp4

https://user-images.githubusercontent.com/122969285/235353763-85aeed76-e9cb-4a27-80e5-9b37f4ce615f.mp4

**Reply by bdach** [reply]
Holding *is* hovering. The drag doesn't start until you start... dragging the slider.

It also isn't clear to me that adjusting settings while a replay ends should stop the replay from progressing to results. The tooltip is perhaps a bit broken and is the closest thing to a "bug" here.

**Reply by monochrome22** [reply]
I didn't know holding intentionally counts as hovering, I thought it was an oversight which allows breaking UX. If it's intentional, then this issue is a duplicate of https://github.com/ppy/osu/issues/6121, and it's correct to have this as a discussion instead.

I still support not progressing the game while holding or dragging, here is my reasoning:
Hovering can happen when the user is just keeping their cursor in a random spot while waiting for the map to start, but if a user is holding a slider then they are not idle, but instead, they are almost certainly thinking about what setting to put. And for replay progressing to results, if the user is adjusting a slider, then it means they are trying to set different settings and they should not be interrupted by replay ending. Holding and hovering should be seen as separate things, or rather, holding should be part of dragging, not hovering.

**Reply by peppy** [reply]
It would definitely be best if we can inhibit progression when interacting with UI controls. I'm not sure how easy it is to implement this (it might have already been if it was easy to achieve, not sure).
